### PR TITLE
feat: add url rule

### PR DESCRIPTION
**Pull request by mike85**
This PR adds the `urlRule` that validates url strings.

### CODE REVIEW

File: packages/sidebar-settings/src/helpers/rules/urlRule.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+export const urlRule: Rule<string> = {
+    errorMessage: 'Please use a valid url',
+    validate: (value: string): boolean => {
+        const regex = new RegExp(
+            '^((http|https)://)[-a-zA-Z0-9@:%._+~#?&/=]{2,256}\\.[a-z]{2,6}\\b([-a-zA-Z0-9@:%._+~#?&/=]*)$',
+            'i',
+        );
+
+        return regex.test(value);
```
Comment (SamuelAlev) [inline packages/sidebar-settings/src/helpers/rules/urlRule.ts:13]:
```suggestion
        try {
          new URL(value)
          return true
        } catch {
           return false
        }
```

Comment (mike85) [reply packages/sidebar-settings/src/helpers/rules/urlRule.ts:13]: This was indeed my first approach but as it considers e.g. `https://index.html` as a valid url, I decided to go the regex approach to be more restrictive. What’s your opinion here?

Comment (SamuelAlev) [reply packages/sidebar-settings/src/helpers/rules/urlRule.ts:13]: This regex approach prevent custom protocols like `frontify://login`

Comment (mike85) [reply packages/sidebar-settings/src/helpers/rules/urlRule.ts:13]: That’s true. So we’ll follow the approach to allow all possible valid urls. Thanks!

**Review by SamuelAlev — APPROVED** [review]

**Pull request merged**